src/alfasim_sdk/_internal/cli.py

```diff
@@ -111,6 +122,17 @@ def template(dst, caption, plugin_id, author_name, author_email):
     python_folder.mkdir()
     Path(python_folder / f"{plugin_id}.py").touch()
 
+    if pyinvoke:
+        # remove compile.py created by hookman
+        compile_file = dst / plugin_id / "compile.py"
+        if compile_file.exists():
+            compile_file.unlink()
```

Comment:
Just an idea.

Given hookman create the `compile.py` file. Isn't it better to incorporate part of that file in the tasks somehow?

Reply:
I don't know. In the discussion in Zulip's topic, we have stablished that when the user chooses to use invoke, the compilation will be done using just the task. But I think your idea is good to support backwards compatibility. What do you think @nicoddemus?

Reply:
I not sure I follow what @prusse-martin is suggesting, can you please clarify it?

Reply:
I believe @prusse-martin is suggesting that we still use the `compile.py` file generated, and the task `compile` will just execute `python compile.py` similarly to what the `alfasim-sdk compile` was doing previously. Am I right?

Reply:
But what are the benefits of that over just following the pattern of using `inv compile` as for all other tasks? Remember that there's no backward compatibility concerns here.

Reply:
AHh OK re-reading I understand the suggestion.

I'm not against the `inv compile` taks just executing `python compile.py` behind the scenes, given `compile.py` is created by hookman.

But does that make sense still, I mean we compile the *plugin* using the `compile.py` unchanged, as (generated by hookman)?

Reply:
See my suggestion, in another comment, I think there's some confusion: we don't need to have two "modes" for new, one legacy and one using "pyinvoke" (with tasks.py) -- `new` should now always just use `tasks.py` and that's it.

Reply:
I think it could give the user the ability to change the compilation steps. For instance, passing a different flag to cmake. But I understand that given we will be using just invoke from now on, the approach would be overwriting the `compile` task instead.

Reply:
> ... `compile.py` unchanged, as generated by hookman?

This was my point (not backward compatibility).
If the are some relevant change in hookman using `inv compile` will not be affected by it, that is why I am suggesting to make `inv compile` deffer the actual compilation to `compile.py`.



> I think it could give the user the ability to change the compilation steps.

If the author really want that OK. He is totally free to that.

> For instance, passing a different flag to cmake.

This should probably be covered by hookman when generating the `compile.py` file (extra flags, override flags), and the sdk could provide support to configure this using the spec file (we are going to add an spec file, or I am mistaken?).

> But I understand that given we will be using just invoke from now on, the approach would be overwriting the `compile` task instead.

`compile.py` it is a generated file, if the author changes it he is at his own (I maybe wrong and hookman's compile.py be supposed to be changed).
On the `inv compile` front I am not totally sure we want to encourage that. Can think of ppl coming up with a lot of fancy ways of creating incompatible plugins.

(BTW, I did not yet read the rest of the comments, sorry if I get a bit redundant)

Reply:
Currently, the users are already changing the `compile.py`. For instance, check our hydrates plugin (I will not link it here, because this is a public repo).

Reply:
As discussed in the daily meeting, we can merge it as it is and further improvement could be made afterwards.